plugin/evm/orderbook_test.go

```diff
@@ -0,0 +1,262 @@
+
+	// orderBookABI, err = abi.FromSolidityJson(string(limitorders.OrderBookAbi))
+	orderBookABI, err = abi.FromSolidityJson(orderBookABIStr)
+	if err != nil {
+		panic(err)
+	}
+
+	aliceKey, _ = crypto.HexToECDSA("56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027")
+	bobKey, _ = crypto.HexToECDSA("31b571bf6894a248831ff937bb49f7754509fe93bbd2517c9c73c4144c0e97dc")
+	alice = crypto.PubkeyToAddress(aliceKey.PublicKey)
+	bob = crypto.PubkeyToAddress(bobKey.PublicKey)
+}
+
+func createPlaceOrderTx(t *testing.T, vm *VM, trader common.Address, privateKey *ecdsa.PrivateKey, size *big.Int, price *big.Int, salt *big.Int) common.Hash {
+	nonce := vm.txPool.Nonce(trader)
+
+	order := limitorders.Order{
+		Trader:            trader,
+		AmmIndex:          big.NewInt(0),
+		BaseAssetQuantity: big.NewInt(0).Mul(size, _1e18),
+		Price:             big.NewInt(0).Mul(price, _1e6),
+		Salt:              salt,
+	}
+	data, err := orderBookABI.Pack("placeOrder", order, []byte("some data"))
+	if err != nil {
+		t.Fatalf("orderBookABI.Pack failed: %v", err)
+	}
+	tx := types.NewTransaction(nonce, orderBookAddress, big.NewInt(0), 8000000, big.NewInt(500000000000), data)
+	signer := types.NewLondonSigner(vm.chainConfig.ChainID)
+	signedTx, err := types.SignTx(tx, signer, privateKey)
+	if err != nil {
+		t.Fatalf("types.SignTx failed: %v", err)
+	}
+	errs := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		if err != nil {
+			t.Fatalf("lop.txPool.AddOrderBookTx failed: %v", err)
+		}
+	}
+	return signedTx.Hash()
+}
+
+//	  A
+//	 / \
+//	B   C
+//	    |
+//	    D (matching tx of order 1 and 2)
+
+// vm1 proposes block A containing order 1
+// block A is accepted
+// vm1 proposes block B containing order 2
+// vm1 and vm2 set preference to block B
+// vm2 proposes block C containing order 2 & order 3
+// vm1 and vm2 set preference to block C
+// vm2 proposes block D containing matching tx of order 1 and 2
+// vm1 and vm2 set preference to block D
+// reorg happens when vm1 accepts block D
```

Comment:
if vm1 has already set preference to block c (line 87), why would a reorg accur when D is accepted (line 90)?

Reply:
Mistake in the comments. Reorg happens when vm1 accepts block C. Updated the comments now